programs/server/Server.cpp

```diff
@@ -977,6 +977,8 @@ if (ThreadFuzzer::instance().isEffective())
             CertificateReloader::instance().tryLoad(*config);
 #endif
             ProfileEvents::increment(ProfileEvents::MainConfigLoads);
+
+            global_context->setConfig(config);
```

Comment:
Some configuration parameters are not supposed to be changed in runtime. Probably there are many places in code that rely on the fact that configuration parameters are not reloaded by default. I think it's not a problem if, for example, all ON CLUSTER queries will fail (see [1](https://github.com/ClickHouse/ClickHouse/blob/49f49da06a1fa9605d6771ba887e398506f50147/src/Interpreters/Context.cpp#L2110-L2116), [2](https://github.com/ClickHouse/ClickHouse/blob/2bf15d6b5b058073304320f4bbe33506ea143f12/src/Interpreters/DDLTask.cpp#L163-L173)) if user change `tcp_port` in config without restarting server. The problem is that we actually do not know what may happen. So we should check all places where `Context::getConfig()`, `Context::getConfigRef()` and `Application::config()` are used (including indirect usages) and make some fixes where needed. But I would prefer to explicitly reload some parts of config that we want to reload and do not think about other parts.